src/ceph-volume/ceph_volume/util/system.py

```diff
@@ -33,21 +33,31 @@ def generate_uuid():
 
 def which(executable):
     """find the location of an executable"""
-    locations = (
+    def _get_path(executable, locations):
+        for location in locations:
+            executable_path = os.path.join(location, executable)
+            if os.path.exists(executable_path) and os.path.isfile(executable_path):
+                return executable_path
+
+    path = os.getenv('PATH', '')
+    path_locations = path.split(':')
```

Comment:
So I actually had to look, if this is actually correct. And it seems, yes: *colon* is [unconditionally and inescapable used as path separator](https://stackoverflow.com/questions/14661373/how-to-escape-colon-in-path-on-unix). But still manually parsing PATH just looks wrong. 

Can we just use 

```python
from distutils.spawn import find_executable
find_executable(executable)  # returns either the absolute path or `None`
```

here?

See also in ceph-daemon:

https://github.com/ceph/ceph/blob/d7821ce8ec6899cbf9f789c8df4d1921c0052e1f/src/ceph-daemon/ceph-daemon#L283-L288

Reply:
I think as long as we need to check other locations besides `$PATH` anyway, its fine as is. `find_executable` does exactly the same thing plus some Windows related handling [1].

Once we can get rid of the additional locations, I'm :+1: on using a library, until then I like the explicitness and it makes sense to use the code we need anyway over importing a library.

[1] https://github.com/python/cpython/blob/62dfd7d6fe11bfa0cd1d7376382c8e7b1275e38c/Lib/distutils/spawn.py#L169